### PR TITLE
feat(eventProcessor): Incorporate batch processing into OptimizelyFactory.

### PR DESCRIPTION
**Pull request by rashidsp**
## Summary
- Incorporate `BatchEventProcessor` with `OptimizelyFactory`.

## Test plan
-  Added unit tests

### CODE REVIEW

**Comment by mjc1283** [comment]
@rashidsp Could this be opened against master after merging #197 and/or #200?

**Comment by rashidsp** [comment]
> @rashidsp Could this be opened against master after merging #197 and/or #200?

against master

File: lib/optimizely/optimizely_factory.rb [hunk 1]
```diff
@@ -57,6 +105,13 @@ def self.custom_instance(
       config_manager = nil,
       notification_center = nil
     )
+      event_processor = BatchEventProcessor.new(
```
Comment (msohailhussain) [inline lib/optimizely/optimizely_factory.rb]: Can you please add in a doc, default batchsize and flush_interval will be used to setup batchEventProcessor.

Comment (rashidsp) [reply lib/optimizely/optimizely_factory.rb]: Done

**Comment by mjc1283** [comment]
@msohailhussain Could you please review the updates from @rashidsp ?

**Pull request merged**